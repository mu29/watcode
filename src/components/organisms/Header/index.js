import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { IconButton } from 'components'
import { SearchBar } from 'containers'
import { palette } from 'services/style'

const color = ({ to, location, theme }) => palette(to === location ? 'yellow.default' : 'gray.30')({ theme })
const borderColor = ({ to, location, theme }) => palette(to === location ? 'yellow.default' : 'transparent.default')({ theme })

const Wrapper = styled.div``

const TopArea = styled.div`
  background-color: ${palette('gray.100')};
  padding: 1rem 0;
`

const BottomArea = styled.div`
  background-color: ${palette('gray.90')};
`

const Container = styled.div`
  display: flex;
  align-items: center;
  max-width: 64rem;
  width: 100%;
  margin: auto;
`

const Image = styled.img`
  width: 2.5rem;
  height: 2.5rem;
  margin: 0 1rem;
`

const StyledIconButton = styled(props => <IconButton { ...props } />)`
  height: 2.5rem;
  padding: 0.125rem 0.5rem 0 0.5rem;
  margin: 0 0.5rem;
  font-size: 0.875rem;
  color: ${color};
  border-bottom: 0.125rem solid ${borderColor};
  border-radius: 0;
  background-color: transparent;

  &:first-child {
    margin-left: 1rem;
  }
`

const Menu = styled.span`
  margin-left: 0.4rem;
`

const Header = ({
  location,
}) => (
  <Wrapper>
    <TopArea>
      <Container>
        <Image src="/resources/logo.svg" />
        <SearchBar />
        <StyledIconButton icon="user" to="/auth" location={ location } />
      </Container>
    </TopArea>
    <BottomArea>
      <Container>
        <StyledIconButton icon="home" to="/" location={ location }>
          <Menu>홈</Menu>
        </StyledIconButton>
        <StyledIconButton icon="star" to="/bookmarks" location={ location }>
          <Menu>즐겨찾기</Menu>
        </StyledIconButton>
        <StyledIconButton icon="comment" to="/community" location={ location }>
          <Menu>커뮤니티</Menu>
        </StyledIconButton>
      </Container>
    </BottomArea>
  </Wrapper>
)

Header.propTypes = {
  location: PropTypes.string.isRequired,
}

export default Header
