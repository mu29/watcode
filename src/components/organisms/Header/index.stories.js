import React from 'react'
import { storiesOf } from '@storybook/react'
import { Header } from 'components'

storiesOf('Organisms/Header', module)
  .add('기본', () => (
    <Header location="/" />
  ))
  .add('즐겨찾기', () => (
    <Header location="/bookmarks" />
  ))
  .add('로그인', () => (
    <Header location="/auth" />
  ))
