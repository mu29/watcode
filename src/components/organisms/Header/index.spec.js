import React from 'react'
import Header from '.'

const attributes = { location: '/' }

const component = (props = attributes) => withTheme(<Header { ...props } />)
const wrap = (props = attributes) => shallow(component(props)).dive()

describe('<Header />', () => {
  it('정상적으로 렌더링된다', () => {
    const wrapper = render(component())
    expect(wrapper).toMatchSnapshot()
  })

  it('5개의 메뉴를 가진다', () => {
    const wrapper = wrap()
    expect(wrapper.find('StyledIconButton')).toHaveLength(5)
    const firstMenu = wrapper.childAt(0).childAt(0).dive().dive()
    expect(firstMenu).toMatchSnapshot()
    const lastMenu = wrapper.childAt(1).childAt(0).dive().dive()
    expect(lastMenu).toMatchSnapshot()
  })

  it('커뮤니티 메뉴가 선택되어 있다', () => {
    const wrapper = wrap({ location: '/community' })
    expect(wrapper).toMatchSnapshot()
  })
})
