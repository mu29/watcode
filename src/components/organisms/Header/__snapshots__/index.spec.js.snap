// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> 5개의 메뉴를 가진다 1`] = `
.c0 {
  margin-bottom: -0.0625rem;
  border-right: 0.0625rem solid;
  border-bottom: 0.0625rem solid;
  border-radius: 0;
}

.c0:first-child {
  border-left: 0.0625rem solid;
}

<IconButton
  className="c0"
  icon="home"
  location="/"
  to="/"
/>
`;

exports[`<Header /> 5개의 메뉴를 가진다 2`] = `
.c0 {
  margin-bottom: -0.0625rem;
  border-right: 0.0625rem solid;
  border-bottom: 0.0625rem solid;
  border-radius: 0;
}

.c0:first-child {
  border-left: 0.0625rem solid;
}

<IconButton
  className="c0"
  icon="user"
  location="/"
  to="/auth"
/>
`;

exports[`<Header /> 정상적으로 렌더링된다 1`] = `
.c0 {
  padding: 1rem 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 64rem;
  width: 100%;
  margin: auto;
}

.c2 {
  width: 2.5rem;
  height: 2.5rem;
  margin: 0 1rem;
}

.c3 {
  margin-left: 0.4rem;
}

<div
  class=""
>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        class="c2"
        src="/resources/logo.svg"
      />
      <span
        componentname="SearchBar"
      />
      <span
        componentname="IconButton"
      />
    </div>
  </div>
  <div
    class=""
  >
    <div
      class="c1"
    >
      <span
        componentname="IconButton"
      >
        <span
          class="c3"
        >
          홈
        </span>
      </span>
      <span
        componentname="IconButton"
      >
        <span
          class="c3"
        >
          즐겨찾기
        </span>
      </span>
      <span
        componentname="IconButton"
      >
        <span
          class="c3"
        >
          커뮤니티
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`<Header /> 커뮤니티 메뉴가 선택되어 있다 1`] = `
<div
  className=""
>
  <TopArea>
    <Container>
      <Image
        src="/resources/logo.svg"
      />
      <SearchBar />
      <StyledIconButton
        icon="user"
        location="/community"
        to="/auth"
      />
    </Container>
  </TopArea>
  <BottomArea>
    <Container>
      <StyledIconButton
        icon="home"
        location="/community"
        to="/"
      >
        <Menu>
          홈
        </Menu>
      </StyledIconButton>
      <StyledIconButton
        icon="star"
        location="/community"
        to="/bookmarks"
      >
        <Menu>
          즐겨찾기
        </Menu>
      </StyledIconButton>
      <StyledIconButton
        icon="comment"
        location="/community"
        to="/community"
      >
        <Menu>
          커뮤니티
        </Menu>
      </StyledIconButton>
    </Container>
  </BottomArea>
</div>
`;
